### PR TITLE
[Merged by Bors] - feat(Order): sup/inf equal to top/bot if top/bot in set

### PR DESCRIPTION
**Pull request by Rida-Hamadani**
The sup lemma is needed in a proof related to `SimpleGraph.ediam` in #12058 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/CompleteLattice.lean [hunk 1]
```diff
@@ -540,6 +540,12 @@ theorem sInf_eq_bot : sInf s = ⊥ ↔ ∀ b > ⊥, ∃ a ∈ s, a < b :=
   @sSup_eq_top αᵒᵈ _ _
 #align Inf_eq_bot sInf_eq_bot
 
+lemma sup_eq_top_of_top_mem (h : ⊤ ∈ s) : sSup s = ⊤ :=
```
Comment (urkud) [inline Mathlib/Order/CompleteLattice.lean:543]: This should be true in a more general case. At least, for a complete lattice. Possibly, for a conditionally complete lattice with top.

Comment (urkud) [reply Mathlib/Order/CompleteLattice.lean:543]: The proof for a conditionally complete lattice + `OrderTop` is `top_unique <| le_csSup (OrderTop.bddAbove s) h`

Comment (Rida-Hamadani) [reply Mathlib/Order/CompleteLattice.lean]: You are right, thank you!

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**